app/map.js

```diff
@@ -0,0 +1,205 @@
+var app = require('remote').require('app');
+var jetpack = require('fs-jetpack').cwd(app.getAppPath());
+
+function buildShaderProgram(gl) {
+    var vertexShader = gl.createShader(gl.VERTEX_SHADER);
+    gl.shaderSource(vertexShader, jetpack.read("../app/vertex.glsl"));
+    gl.compileShader(vertexShader);
+    if (!gl.getShaderParameter(vertexShader, gl.COMPILE_STATUS)) {
+        throw new Error("Failed to compile vertex shader: " + gl.getShaderInfoLog(vertexShader));
+    }
+
+    var fragmentShader = gl.createShader(gl.FRAGMENT_SHADER);
+    gl.shaderSource(fragmentShader, jetpack.read("../app/fragment.glsl"));
+    gl.compileShader(fragmentShader);
+    if (!gl.getShaderParameter(fragmentShader, gl.COMPILE_STATUS)) {
+        throw new Error("Failed to compile fragment shader: " + gl.getShaderInfoLog(fragmentShader));
+    }
+
+    var program = gl.createProgram();
+    gl.attachShader(program, vertexShader);
+    gl.attachShader(program, fragmentShader);
+    gl.linkProgram(program);
+    if (!gl.getProgramParameter(program, gl.LINK_STATUS)) {
+        throw new Error("Failed to link shader program: " + gl.getProgramInfoLog(program));
+    }
+
+    return program;
+}
+
+function buildTileDataTexture(data) {

```

Comment:
When I was working on it I had a thought that you might be able to do it with an attribute array, the same way that the position data is passed in. However, textures seem to be the best way to handle very large arrays of homogeneous data like this, so I just stuck with the texture for now.